### PR TITLE
Add option to enter endpoint URL (fix #116)

### LINKED ISSUE #116: Add option to enter endpoint URL
Currently, https://apis.guru/graphql-voyager/ supports visualizing GraphQL APIs by
1. entering SDL
2. entering introspection
3. using one of built-in samples (Star Wars, Yelp, Shopify Storefront, GitHub)

But it doesn't support visualizing GraphQL APIs by entering the GraphQL API endpoint URL.

This is the first thing I was looking for when I first saw a GraphQL Voyager to try it with my API (with minimal effort). Entering URL is the most intuitive and expected way for users. BTW, visualizing by URL is already implemented in https://birdseye.novvum.io and https://brbb.github.io/graphql-rover/.

### PR DESCRIPTION
**Pull request by FluorescentHallucinogen**

### CODE REVIEW

File: demo/IntrospectionModal.tsx [hunk 1]
```diff
@@ -156,12 +188,33 @@ export class IntrospectionModal extends React.Component<IntrospectionModalProps>
...
+    const { urlText, headers } = this.state.current;
+    return (
+      <>
+        <textarea
+          value={urlText || ''}
+          placeholder="Paste URL here"
+          onChange={this.handleURLChange}
+        />
+        <div>
+          Paste headers (in JSON format) into the textarea below.
```
Comment (IvanGoncharov) [inline demo/IntrospectionModal.tsx]: Great for PoC but it bad UX in the long run, it would be constant source of frustration for users.

File: demo/IntrospectionModal.tsx [hunk 2]
```diff
@@ -64,12 +67,27 @@ export class IntrospectionModal extends React.Component<IntrospectionModalProps>
...
+
   handleSubmit = () => {
-    const { inputType, activePreset, jsonText, sdlText } = this.state.current;
+    const { inputType, activePreset, urlText, headers, jsonText, sdlText } = this.state.current;
     switch (inputType) {
       case Presets:
         this.props.onChange(PRESETS[activePreset]);
         break;
+      case URL:
+        this.fetchIntrospection(urlText, headers).then((data) => this.props.onChange(data));
```
Comment (IvanGoncharov) [inline demo/IntrospectionModal.tsx]: How UI look like during fetch?
and where is error handling?

File: demo/IntrospectionModal.tsx (hunk 1 above)
Comment (FluorescentHallucinogen) [reply demo/IntrospectionModal.tsx]: I believe this is an acceptable solution. Please take a look at GraphQL Playground. Users paste headers in GraphQL Playground in JSON format. GraphQL Playground is production-ready tool.

graphqlbin

Also users paste variables in GraphiQL and GraphQL Playground in JSON format too. The target audience of these tools are developers. So I think this is ok.

File: demo/IntrospectionModal.tsx (hunk 2 above)
Comment (FluorescentHallucinogen) [reply demo/IntrospectionModal.tsx]: UI looks the same as for the rest "Introspection" and "SDL" tabs. Please note, there is no any error handling for them too. E.g.:

* If the user has entered invalid JSON or SDL into textarea, there is no any UI toast about JSON parsing or introspection from schema building errors.
* If the user has not entered JSON or SDL into textarea at all, the "Display" button is still active.

Errors are printed in the browser console.

**Comment by FluorescentHallucinogen** [comment]
@IvanGoncharov Thanks for your comments! They are all great points. But I don't plan to fix them as part of this pull request because they are out of scope.

I'm guided by [Google's Engineering Practices](https://google.github.io/eng-practices/review/reviewer/standard.html).

TL;DR: The product/code should get better than before merging a change, even if it isn’t perfect.

Before merging my change, users couldn't enter GraphQL endpoint URLs at all. After merging, they could. So the product is in better state than before. The quality of my change is not lower than the quality of the code. So it doesn't degrade the overall code health.

Nevertheless, I would be happy to fix them (and some other UI/UX issues I found) in the next iteration as a separate pull request. :wink:

**Comment by FluorescentHallucinogen** [comment]
Deployed a live demo: https://graphql-voyager.web.app.

**Comment by FluorescentHallucinogen** [comment]
Updated live demo: https://graphql-voyager.web.app.

graphql-voyager-url-tab

**Comment by ravisastryk** [comment]
This is an excellent feature that helps navigate the graph especially when there are multiple services which are federated. We have a use case where we would like to specify URL and let the voyager render the graph. Appreciate if this can be merged.

**Comment by nohehf** [comment]
Hey @IvanGoncharov, what's the state of this feature request? If the MR is too old to be merged I'll be happy to open a new mr with those changes (and we can discuss which changes to implement). 
I think this feature is a great UX / quality of life improvement to the product and should be considered.
